extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisor.java

```diff
@@ -0,0 +1,507 @@
+/*
+        reportingExec.scheduleAtFixedRate(
+            computeAndEmitLag(taskClient),
+            ioConfig.getStartDelay().getMillis() + 10000, // wait for tasks to start up
+            Math.max(monitorSchedulerConfig.getEmitterPeriod().getMillis(), 60 * 1000),
+            TimeUnit.MILLISECONDS
+        );
+*/
+  }
+
+  @Override
+  protected int getTaskGroupIdForPartition(String partitionId)
+  {
+    if (!partitionIds.contains(partitionId)) {
+      partitionIds.add(partitionId);
+    }
+
+    return partitionIds.indexOf(partitionId) % spec.getIoConfig().getTaskCount();
+  }
+
+  @Override
+  protected boolean checkSourceMetadataMatch(DataSourceMetadata metadata)
+  {
+    return metadata instanceof KinesisDataSourceMetadata;
+  }
+
+  @Override
+  protected boolean doesTaskTypeMatchSupervisor(Task task)
+  {
+    return task instanceof KinesisIndexTask;
+  }
+
+  @Override
+  protected SeekableStreamSupervisorReportPayload<String, String> createReportPayload(
+      int numPartitions,
+      boolean includeOffsets
+  )
+  {
+    KinesisSupervisorIOConfig ioConfig = spec.getIoConfig();
+    return new KinesisSupervisorReportPayload(
+        spec.getDataSchema().getDataSource(),
+        ioConfig.getStream(),
+        numPartitions,
+        ioConfig.getReplicas(),
+        ioConfig.getTaskDuration().getMillis() / 1000,
+        spec.isSuspended()
+    );
+  }
+
+  // not yet supported, will be implemented in the future
+  @Override
+  protected Map<String, String> getLagPerPartition(Map<String, String> currentOffsets)
+  {
+    return ImmutableMap.of();
+  }
+
+  @Override
+  protected SeekableStreamDataSourceMetadata<String, String> createDataSourceMetaData(
+      String stream,
+      Map<String, String> map
+  )
+  {
+    return new KinesisDataSourceMetadata(
+        new SeekableStreamPartitions<>(stream, map)
+    );
+  }
+
+  @Override
+  protected OrderedSequenceNumber<String> makeSequenceNumber(
+      String seq,
+      boolean isExclusive
+  )
+  {
+    return KinesisSequenceNumber.of(seq, isExclusive);
+  }
+
+  // the following are for unit testing purposes only
+  @Override
+  @VisibleForTesting
+  protected void runInternal()
+      throws ExecutionException, InterruptedException, TimeoutException, JsonProcessingException, NoSuchMethodException,
+             IllegalAccessException, ClassNotFoundException
+  {
+    super.runInternal();
+  }
+
+  @Override
+  @VisibleForTesting
+  protected Runnable updateCurrentAndLatestOffsets()
+  {
+    return super.updateCurrentAndLatestOffsets();
+  }
+
+  @Override
+  protected void updateLatestSequenceFromStream(
+      RecordSupplier<String, String> recordSupplier, Set<StreamPartition<String>> streamPartitions
+  )
+  {
+    // do nothing
+  }
+
+  @Override
+  @VisibleForTesting
+  protected void gracefulShutdownInternal() throws ExecutionException, InterruptedException, TimeoutException
+  {
+    super.gracefulShutdownInternal();
```

Comment:
No need to implement this method.

Reply:
There're several methods that you don't have to implement. Please remove them all.